src/main/java/net/mcreator/generator/GeneratorTemplatesList.java

```diff
@@ -34,13 +30,16 @@
  *
  * @param groupName The name of this group of templates shown in workspace panel.
  * @param listData  The collection used by workspace to generate given templates.
- * @param element   The {@link GeneratableElement} used to process tokens in template names.
- * @param templates The map of templates to be generated for each entry of {@code listData};
- *                  keys are templates themselves, values represent generation conditions of their key template
- *                  for all items on the mentioned collection.
+ * @param templates The list of templates to be generated for each {@code listData} item.
  */
-public record GeneratorTemplatesList(String groupName, List<?> listData, GeneratableElement element,
-									 Map<GeneratorTemplate, List<Boolean>> templates) {
+public record GeneratorTemplatesList(String groupName, List<?> listData, List<List<ListTemplate>> templates) {
+
+	/**
+	 * The sole constructor.
+	 */
+	public GeneratorTemplatesList {
+		templates.forEach(l -> l.forEach(e -> e.setTemplatesList(this)));

```

Comment:
This is a circular reference, also where is not guarantee ListTemplate will be linked to GeneratorTemplatesList? One can create ListTemplate out of thin air and it will never require a link to GeneratorTemplatesList as it did before

Reply:
So should it go the same way as `ModAPI` back in #2313?

Reply:
I fail to see resemblance, but as ListTemplate having reference to GeneratorTemplatesList is a core part of the system, having setter to set GeneratorTemplatesList later is not the right approach I would say.

> One can create ListTemplate out of thin air and it will never require a link to GeneratorTemplatesList as it did before

Reply:
I made `ModAPIManager` first create `ModAPI` object that is the parent to `ModAPIImplementation`s created in the loop following, then add those children to the `implementations` map in `ModAPI`. Do I do the same here?

Reply:
That could work, but then make sure ListTemplate is protected and thus created within this system only

Reply:
✔️ Fixed